src/platform/Linux/PlatformManagerImpl.cpp

```diff
@@ -187,6 +278,59 @@ CHIP_ERROR PlatformManagerImpl::_GetCurrentHeapHighWatermark(uint64_t & currentH
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR PlatformManagerImpl::_GetRebootCount(uint16_t & rebootCount)
+{
+    uint32_t count = 0;
+
+    CHIP_ERROR err = ConfigurationMgrImpl().GetRebootCount(count);
+
+    if (err == CHIP_NO_ERROR)
+        rebootCount = static_cast<uint16_t>(count);
```

Comment:
I guess my question is: if the stored reboot count can be larger than uint16_t, are we not losing data here?  And if it can't, why does the configuration manager return uint16_t?

Reply:
Since we only have the following implementations in our low level APIs
    static CHIP_ERROR ReadConfigValue(Key key, bool & val);
    static CHIP_ERROR ReadConfigValue(Key key, uint32_t & val);
    static CHIP_ERROR ReadConfigValue(Key key, uint64_t & val);
   
Configuration mgr APIs are aligned with ReadConfigValue/WriteConfigValue, otherwise, we need to do the same conversion within configuration mgr APIs.

Reply:
Ah, OK.  In that case, I guess it's just deciding whether this should error out if the count is too high or let it roll over... Worth a comment either way explaining the decision.

Reply:
Acked. I's prefer return error out in this case.